pkg/controller/node/controller_utils.go

```diff
@@ -36,6 +35,19 @@ import (
 	"github.com/golang/glog"
 )
 
+func ComputeZoneState(nodeConditions []*api.NodeCondition) zoneState {
```

Comment:
Please add explicit comment that you assume that those are only "Ready" conditions (and name variable like that).